### ISSUE TITLE
question: need to correct MD/NM/UQ tags multiple times on piped commands or not?

### ISSUE THREAD

**Issue by eboyden** [fulcrumgenomics/fgbio#877]
Hi, I'm wondering if fgbio tools that modify reads (e.g. TrimPrimers, ClipBam) require correct MD/NM/UQ tags to function properly, and therefore they need to be correct prior to each call?  So if several commands are piped in a row (e.g. `TrimPrimers | ClipBam`) they should be corrected at every step?  Or is it safe to only correct them at the final step to save time?  Thanks!

**Comment by eboyden** [comment]
Related: specifying a reference for recalculating these tags is optional for TrimPrimers, but required for ClipBam.  If correct tags aren't strictly required for the tools to operate properly, it would be nice to make it optional for ClipBam as well (with a warning in the doc) so that extra sorting isn't required (e.g. if tag correction will be performed downstream).

**Comment by nh13** [comment]
@eboyden would you be willing to sponsor some of this work to improve these tools?

**Comment by eboyden** [comment]
I'm not opposed but I'm working on behalf of my company (Molecular Loop Biosciences) - I will bring it up.  In the meantime please understand that these are questions and suggestions, not demands!  If time doesn't allow working on them soon (or ever), I understand (and am grateful for this software being open source at all).

**Comment by nh13** [comment]
Will do.  We're always looking for help improving our tools, whether that's from pull requests or sponsorship.  If that's an option for your company, we'd be glad to chat: https://www.fulcrumgenomics.com/

**Comment by nh13** [comment]
I don't think we use the MD/NM/UQ tags much at all in fgbio, so really it's more of a concern if tools that consume the output of fgbio use those tags.  We do our best to either update those tags or invalidate (remove) them so the outputs are valid, since we want to keep fgbio general purpose.  I can see having an optimized version that skips steps like these.  A quick scan of the repo shows the following:

|Tool|Behavior|
|---|---|
|`CallOverlappingConsensusBases`|update|
|`ClipBam`|update|
|`FilterConsensusReads`|update|
|`RemoveSamTags`|remove|
|`TrimPrimers`|remove (or update if a FASTA is given)|
